### PR TITLE
Tutorials: Final steps of march tutorial 

### PR DESCRIPTION
**Pull request by musicEnfanthen**
This PR goes on tutorial branch and is ready to _review_. 

Please do not simply merge, but let's use this PR to improve it further together.

The PR provides the basic structure with all needed files, steps, instructions and xpath rules for the advanced tutorial (march) developed at #MEIOx . It also moves the individual steps into separate subfolders for a more readable structure. It does **not** change the main logic (tutorials.js), view (tutorials.html) or style (CSS) in regard of @kepper working on it at the moment.

It certainly still needs the proofreading of a native speaker (would you like to have a look @siennamw ? Everyone else is invited to take a look at it, too.)

@soniawronkowska @margrethesb could you please have a look especially at the proposed` "...-desc.html"` files here where all the instructions are included and give your feedback here via GitHub's review function? I had to stretch the tutorial to 12 steps in favor of having a better distinction of several issues.

@kepper could you (beside everythig else ;)) have a closer look at the xpath rules provided and the correctness of the content information in the instructions?

Looking forward to your feedback.

### CODE REVIEW

File: _tutorials/102_march/step-11/102_march_step-11-desc.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<div>
+    <p id="tutorialGoal">In this step, you will add the dynamic marking to your encoding of the example.</p>
+
+    <p>You define dynamics with a <code>&lt;dynam&gt;</code> element. The <code>&lt;dynam&gt;</code> element may be used for instantaneous or continuous textual dynamics, e.g. <code>“p”</code>, <code>“mf”</code>, or <code>“cresc. poco a poco”</code>.</p>
+
+    <p>Unlike articulation signs, the <code>&lt;dynam&gt;</code> element is a so called <code>control event</code> which groups elements (such as dynamics, ties, phrase marks, pedal marks, etc.), which depend upon other events, such as notes or rests, for their existence. In general, <code>control events</code> are not encoded as child elements of the corresponding events they are meant to control. In practice, it means that they are placed outside the <code>staff</code> elements in most cases.</p>
```
Comment (ahankinson) [inline _tutorials/102_march/step-11/102_march_step-11-desc.html]: "so called" -> "so-called"

File: _tutorials/102_march/step-12/102_march_step-12-desc.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<div>
+    <p id="tutorialGoal">Well done, you'are almost there. In this step, you will add the last missing detail of the example: the slur in the third measure.</p>
```
Comment (ahankinson) [inline _tutorials/102_march/step-12/102_march_step-12-desc.html]: "You'are" -> "You are"

File: _tutorials/102_march/step-12/102_march_step-12-desc.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    <p>There are two different ways, to encode a slur in MEI:</p>
```
Comment (ahankinson) [inline _tutorials/102_march/step-12/102_march_step-12-desc.html]: no comma between 'ways' and 'to'

File: _tutorials/102_march/step-12/102_march_step-12-desc.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <ol>
+        <li>You can use a <code>@slur</code> attribute on <code>&lt;note&gt;</code> or <code>&lt;chord&gt;</code> with a value of <code>i=initial</code>, <code>m=medial</code> or <code>t=terminal</code>.</li>
+        <li>You can use a separate <code>&lt;slur&gt;</code> element.</li>
+    </ol>
+
+    <p>In this tutorial step, you will use the second approach: Like dynamics, <code>&lt;slur&gt;</code> elements are control events, too, what means, that they are not encoded as child elements of the corresponding events they are meant to control but in most cases outside the <code>staff</code> elements.</p>
```
Comment (ahankinson) [inline _tutorials/102_march/step-12/102_march_step-12-desc.html]: This sentence should be broken into two sentences. "...elements are control events. They are not encoded..."

File: _tutorials/102_march/step-12/102_march_step-12-desc.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    <p>To indicate the starting point and end point of the slur, it is possible to apply a reference to the corresponding <code>@xml:id</code> of a <code>&lt;note&gt;</code> or <code>&lt;chord&gt;</code> element. <code>@xml:id</code>'s are identifiers with an arbitrary but unique character sequence that regularize the naming of an element throughout a document and thus facilitate building links between elements and other resources. For <code>&lt;slur&gt;</code>s, a reference to the <code>@xml:id</code> of an element can be given via the attributes <code>@startid</code> and <code>@endid</code>.</p>
```
Comment (ahankinson) [inline _tutorials/102_march/step-12/102_march_step-12-desc.html]: No apostrophe-s after 'xml:id'

File: _tutorials/102_march/step-12/102_march_step-12-desc.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    <p>Along with <code>@startid</code> and <code>@endid</code>, you are supposed to use the following attributes in this step:</p>
```
Comment (ahankinson) [inline _tutorials/102_march/step-12/102_march_step-12-desc.html]: "are supposed to" -> "should"

File: _tutorials/102_march/102_March.md [hunk 7]
```diff
@@ -2,9 +2,11 @@
...
+fullname: "An advanced tutorial for MEI encoding"
 data: "102_march.json"
 ---
 Welcome! In this tutorial you will learn how to encode the following music example in MEI:
 
-"MarchExample"
\ No newline at end of file
+![Hallingdal Bataljons Marsch](../102_march.png)
+
+By the way: this is the beginning of _Hallingdal Bataljons Marsch_, the Opus 1 of the norwegian composer  [Johan Halvorsen](https://de.wikipedia.org/wiki/Johan_Halvorsen) (1864–1935).
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/102_March.md]: Done

File: _tutorials/102_march/102_march.json [hunk 8]
```diff
@@ -8,9 +8,12 @@
       "xmlFile": "102_march_step-01.xml",
       "xpaths": [
         {"rule": "count(//mei:scoreDef) = 1", "renderanyway": false, "hint": "You need one scoreDef element."},
+        {"rule": "count(//mei:scoreDef/@*) = 0", "renderanyway": false, "hint": "You need no attribute on the scoreDef element."},
         {"rule": "count(//mei:staffGrp) = 1", "renderanyway": false, "hint": "You need one staffGrp element."},
+        {"rule": "count(//mei:staffGrp/@*) = 0", "renderanyway": false, "hint": "You need no attribute on the staffGrp element."},
         {"rule": "//mei:scoreDef/mei:staffGrp", "renderanyway": false, "hint": "staffGrp element has to be a child of scoreDef element."},
         {"rule": "count(//mei:staffDef) = 1", "renderanyway": false, "hint": "You need one staffDef element."},
+        {"rule": "count(//mei:staffDef/@*) = 0", "renderanyway": false, "hint": "You need no attribute on the staffDef element."},
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/102_march.json]: Done

File: _tutorials/102_march/step-06/102_march_step-06-desc.html [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<div>
+    <p id="tutorialGoal">In this step you will encode a beamed group of notes.</p>
+
+    <p>To group notes under a beam within a measure, the <code>&lt;beam&gt;</code> element is to be used. The <code>&lt;beam&gt;</code> element is a container for a series of explicitly beamed events that begins and ends entirely within a measure.</p>
+
+    <p id="tutorialTask">In the editor below, please add a <code>&lt;beam&gt;</code> element at the specified position before adding the two child <code>&lt;note&gt;</code>'s separately (e flat and g).</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-06/102_march_step-06-desc.html]: Done

File: _tutorials/102_march/step-07/102_march_step-07-desc.html [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<div>
+    <p id="tutorialGoal">In this step you will learn to encode dotted durations.</p>
+
+    <p>To encode a dotted rhythm, you need to use a <code>@dots</code> attribute on <code>&lt;note&gt;</code> element and set the number of dots as the attribute's value, e.g. <code>“1”</code> or <code>“2”</code>.</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-07/102_march_step-07-desc.html]: Done

File: _tutorials/102_march/step-07/102_march_step-07-desc.html [hunk 11]
```diff
@@ -0,0 +1,7 @@
+<div>
+    <p id="tutorialGoal">In this step you will learn to encode dotted durations.</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-07/102_march_step-07-desc.html]: Done

File: _tutorials/102_march/step-12/102_march_step-12-desc.html (hunk 2 above)
Comment (musicEnfanthen) [reply _tutorials/102_march/step-12/102_march_step-12-desc.html]: Done

File: _tutorials/102_march/step-12/102_march_step-12-desc.html (hunk 3 above)
Comment (musicEnfanthen) [reply _tutorials/102_march/step-12/102_march_step-12-desc.html]: Done

File: _tutorials/102_march/step-12/102_march_step-12-desc.html (hunk 4 above)
Comment (musicEnfanthen) [reply _tutorials/102_march/step-12/102_march_step-12-desc.html]: Done

File: _tutorials/102_march/step-12/102_march_step-12-desc.html (hunk 6 above)
Comment (musicEnfanthen) [reply _tutorials/102_march/step-12/102_march_step-12-desc.html]: Done

File: _tutorials/102_march/step-08/102_march_step-08-desc.html [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<div>
+    <p id="tutorialGoal">Ok, let's encode the complete next measure to repeat all the things you've learned in the previous steps.</p>
+
+    <p id="tutorialTask">Please encode the next measure (<code>n=2</code>) in the editor below. Add a staff and layer element which contains 4 child elements: a quarter <code>&lt;note&gt;</code>, a <code>&lt;beam&gt;</code> with to eighth <code>&lt;note&gt;</code>'s and a dotted quarter <code>&lt;note&gt;</code> with an eighth <code>&lt;note&gt;</code>.</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-08/102_march_step-08-desc.html]: Done

File: _tutorials/102_march/step-09/102_march_step-09-desc.html [hunk 13]
```diff
@@ -0,0 +1,9 @@
+<div>
+    <p id="tutorialGoal">You should be prepared now to encode the note values of the final two measures of the example.</p>
+
+    <p>Ignore the additional elements, such as dynamics, slurs and accents at this point as they will be adressed in the next steps separately.</p>
+
+    <p>The pause is encoded with a <code>&lt;rest&gt;</code> element having a <code>@dur</code> attribute analogue to the <code>&lt;note&gt;</code> element.</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-09/102_march_step-09-desc.html]: Done

File: _tutorials/102_march/step-09/102_march_step-09-desc.html [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+
+    <p id="tutorialTask">Please encode the next measure (<code>n=2</code>) in the editor below. Add a staff and layer element which contains 4 child elements: a quarter <code>&lt;note&gt;</code>, a <code>&lt;beam&gt;</code> with to eighth <code>&lt;note&gt;</code>'s and a dotted quarter <code>&lt;note&gt;</code> with an eighth <code>&lt;note&gt;</code>.</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-09/102_march_step-09-desc.html]: Removed. Tutorial task was a copy & paste error from step 8.

File: _tutorials/102_march/step-10/102_march_step-10-desc.html [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<div>
+    <p id="tutorialGoal">In this step, you will add the articulation signs to your encoding of the example.</p>
+
+    <p>Articulations, i.e. indications of how to play a note or chord, can be encoded with <code>&lt;artic&gt;</code> elements that are applied as children to <code>&lt;note&gt;</code> elements. From a lot of attributes that are possible with <code>&lt;artic&gt;</code>, you are supposed to use the following in this tutorial step:</p>
```
[the remark this answers is not in the document]
Comment (musicEnfanthen) [reply _tutorials/102_march/step-10/102_march_step-10-desc.html]: Changed to: "You should use the following attributes on `artic` in this tutorial step"

Better now?

File: _tutorials/102_march/step-11/102_march_step-11-desc.html (hunk 1 above)
Comment (musicEnfanthen) [reply _tutorials/102_march/step-11/102_march_step-11-desc.html]: Done

**Comment by ahankinson** [comment]
Do you think this is ready to merge? Any other reviewers want to chime in?

**Comment by musicEnfanthen** [comment]
Just let's wait a little so that the other requested reviewers have time to take a look. 

But feel free to approve (or deny) the latest changes I made after your review meanwhile. ;)

The latest two commits this morning are just updating the PR with the reverted commit on tutorial branch.

**Comment by kepper** [comment]
Btw., @lpugin has found a way to validate against RelaxNG on the client. I don't know how performant that is, and it surely creates less helpful error messages, but together with an MEI-all-anystart.rng, this could greatly simplify the writing of tutorials and reduce the number of xpaths…

**Comment by lpugin** [comment]
Yes: https://jsfiddle.net/lpugin/rmc18zkw/17/
It should be no problem to use anystart.rng . I will test it and see if we can get it integrated in the jekyll website.

**Comment by musicEnfanthen** [comment]
Thanks for all your feedback. You adressed a lot of different issues that are very important. 

But as this PR is only about the content of march tutorial, I suggest to keep things separated to not overload this PR with different issues. So let's discuss the visual appearance of the editor window in #64 and about simplifying file validation in another issue (I'll be happy to refactor/adapt it for the march tutorial in another PR if necessary).

@kepper I really understand your concern about this tutorial being "extreme" but I'd like to understand this rather as a compliment and as proof of concept. ;) It even works for a quite elaborated example. 

Of course, this tutorial is not intended (and never was) to be the first one that a real absolute beginner is supposed to be confronted with. It is rather for "advanced beginners" that have gone through different simpler tutorials which adress the different topics of the march tutorial in separate "single topic" lectures. I would imagine to have links to these "single topic" tutorials from the different steps of the march tutorial  (How to write a note, How to write a beam, How to encode scoreDef?). 

The good thing about having the "extreme", or extended" tutorial first, is that we now can just break it down into smaller learning chunks, using a lot of the already existing code or descriptions. I will be happy to work on these, if you agree that this could be a way to go.

@margrethesb told me that she can have a look at this PR in the next days. So I would like to give her a chance to give some feedback before merging this PR. 

Would that be OK with everyone?

**Comment by margrethesb** [comment]
Sorry for my late reply, GitHub decided to deny my existence for a few days... Anyway, here goes:
1. Merging: Go ahead.
2. Tutorial: This needs more work. I'm not going to try to make suggestions related to the encoding issues, but there are several other things that should be reviewed. Things like lanuage/wording, pedagogy/target group, level of tutorial, whether it should be connected to an gentle text-based intro partly based on older material.
I'm happy to participate in this work, and with the aim of having it ready for MEC.

**Comment by musicEnfanthen** [comment]
Thanks @margrethesb . 

If I understand correctly, this PR is ready to be merged now, and further work will be done in the next PR's. So anyone with a license to merge is welcome to do so :)

**Comment by ahankinson** [comment]
Thank you everyone! This is fantastic.

Once we kick the tires and make sure it all works, an e-mail message to MEI-L from the 'tutorials group' would be brilliant. I think this also illustrates the value of having a work-focused meeting in the Autumn to help move community initiatives forward.

Once again, bravo @musicEnfanthen, @margrethesb and @soniawronkowska .

**Comment by musicEnfanthen** [comment]
> Once again, bravo @musicEnfanthen, @margrethesb and @soniawronkowska .

Thanks, but not to forget mastermind @kepper who, beside everything else, provided idea and programmatical logic for all of this. Thanks a lot!

**Comment by musicEnfanthen** [comment]
As already mentioned in my comment to #66, tutorials are not ready to be included on `master`. Once they can be released from `tutorials` to `master` branch, we will be happy to provide an email message to the community.

**Pull request merged**